src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
Okay, yeah, since we just need to be sure it's called once, and not *when* it's called, you're right!